details/impersonation.md

```diff
@@ -0,0 +1,144 @@
+
+Detailed instruction how to define Test Settings can be find [here](../examples/impersonation/test-settings-ui.md).
+
+Copying Main Settings as Test Settings is not enough. We also have to instruct ROR which users can be considered as impersonators (the ones, who are allowed to impersonate any other user). The list of allowed impersonators and their permissions are defined in the `impersonation` section in ROR Settings:
+
+```yaml
+readonlyrest:
+  
+  access_control_rules:
+
+    [...]
+
+  impersonation:
+
+    - impersonator: admin1      // Who can impersonate? (user name or pattern)
+      users: ["*"]              // Who can be impersonated? (User names or patterns)
+      auth_key: admin1:pass     // Authentication rule required to impersonate (any authentication rule can be used here)
+
+    - impersonator: admin2
+      users: ["dev2"]
+      ldap_authentication: "ldap1"
+```
+
+In the example above, we see that we have two impersonators: `admin1` and `admin2`. The first one can impersonate any user (`*`) and they are able to authenticate using basic auth (`admin1:pass`). The second impersonator can impersonate only `dev2` user. They will be authenticated using `ldap1` connector.
+
+When an impersonator passes wrong credentials ROR will tell Kibana that impersonation is not allowed.
+
+#### Defining mocks of the external services (optional)
+
+ROR has many sophisticated authentication & authorization methods. Some of them are based on external systems like LDAP. A problem with such systems according to the impersonation feature is that, the systems don't support it by default, or the configuration is difficult, or they don't support it at all. 
+
+That's why we decided to solve it totally differently - using mocks. [Wikipedia](https://en.wiktionary.org/wiki/mock) defines `mock` as `an imitation, usually of lesser quality.` And in case of external authentication system we are going provide an imitation of it that tell ACL what users can be authenticated by it. When we consider an authorization service, a mock of it will return the ACL users with their roles in the service. And this is enough for ROR to support impersonation. 
+
+How does ROR use the mocks? Let's suppose we have `ldap_auth` rule. When ROR processes the rule, it:
+* asks a given LDAP if the username can be authenticated with a given password, and if they can ...
+* asks LDAP to list what groups the user belongs to
+
+In the impersonation case, it looks pretty much the same. Just we won't call any LDAP server, rather, the mock will provide the required information (no passwords required):
+* asks the mock if the username exists, and if it does ...
+* asks the mock to tell what groups the user belongs to
+
+**⚠️ IMPORTANT:** When some external service is not mocked, ROR might return to Kibana information that the impersonation is not supported. It's better to always define all mocks, to avoid the "not supported impersonation" ES response.
```

Comment:
- `If one or more of the external services are not mocked, ROR might inform Kibana that the impersonation is not supported.`
- `... to avoid the "Impersonation not supported" ES response`